### PR TITLE
JDK-8344303 : Remove usage of URLUtil.getConnectPermission from sun.awt.SunToolkit and sun.awt.image.URLImageSource

### PR DESCRIPTION
**Pull request by honkar-jdk**
sun.awt.SunToolkit and sun.awt.image.URLImageSource are calling sun.net.util.URLUtil.getConnectPermission(URL)
With the SecurityManager gone there's no need to call this method anymore. 

There are other checkPermissions in these 2 files - SunToolkit, URLImageSource but I have removed only those that are related to URLUtil.getConnectPermission(URL). Other checkPermission removal will be handled in a separate issue.

Integrating this PR will allow to proceed with a dependent JBS issue - [JDK-8344180](https://bugs.openjdk.org/browse/JDK-8344180)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344303](https://bugs.openjdk.org/browse/JDK-8344303): Remove usage of URLUtil.getConnectPermission from sun.awt.SunToolkit and sun.awt.image.URLImageSource (**Sub-task** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22330/head:pull/22330` \
`$ git checkout pull/22330`

Update a local copy of the PR: \
`$ git checkout pull/22330` \
`$ git pull https://git.openjdk.org/jdk.git pull/22330/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22330`

View PR using the GUI difftool: \
`$ git pr show -t 22330`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22330.diff">https://git.openjdk.org/jdk/pull/22330.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22330#issuecomment-2494525307)
</details>

### CODE REVIEW

**Comment by prrace** [comment]
so what about this ?

--- a/src/java.base/share/classes/module-info.java
+++ b/src/java.base/share/classes/module-info.java
@@ -291,7 +291,6 @@
         java.security.jgss,
         jdk.naming.dns;
     exports sun.net.util to
-        java.desktop,
         java.net.http,
         jdk.jconsole,
         jdk.sctp;

File: src/java.desktop/share/classes/sun/awt/image/URLImageSource.java [hunk 1]
```diff
@@ -40,21 +39,7 @@ public class URLImageSource extends InputStreamImageSource {
     int actualPort;
 
     public URLImageSource(URL u) {
-        @SuppressWarnings("removal")
-        SecurityManager sm = System.getSecurityManager();
-        if (sm != null) {
-            try {
-                java.security.Permission perm =
-                      URLUtil.getConnectPermission(u);
```
Comment (azvegint) [inline src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:48]: `URLUtil.getConnectPermission(u);` is no longer used anywhere and can be removed


https://github.com/openjdk/jdk/commit/8c644ca310c44995a1fc899eabb5611b79873c68#diff-50e0953b57d7ba2dbb8faa9b1766b99234d5ae4cb271c48ece13b9e3d0c9ca89

so the jdk/src/java.base/share/classes/sun/net/util/URLUtil.java part can be reverted

Comment (honkar-jdk) [reply src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:48]: Will changes to java.base files - URLUtil.java be handled in [JDK-8344180](https://bugs.openjdk.org/browse/JDK-8344180)? Or do we do it here?

**Comment by honkar-jdk** [comment]
> so what about this ?
> 
> --- a/src/java.base/share/classes/module-info.java +++ b/src/java.base/share/classes/module-info.java

Thank you. Updated. I didn't know we had to remove the linkage to java.desktop here and it is a quick way to check if build works and there are no left over dependencies sun.net.util in java.desktop.

File: src/java.desktop/share/classes/sun/awt/image/URLImageSource.java (hunk 1 above)
Comment (azvegint) [reply src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:48]: > Will changes sun/net/util/URLUtil.java be handled in [JDK-8344180](https://bugs.openjdk.org/browse/JDK-8344180)? 

It has only two subtasks with mentions of `URLUtil`:

- [JDK-8344221](https://bugs.openjdk.org/browse/JDK-8344221) it is already resolved
- this issue

So I guess this removal can be the part of this PR

Comment (honkar-jdk) [reply src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:48]: @prrace There is no other code using it, the 2 awt.sun files were the only two files that were using URLUtil.getConnectPermission().

@dfuch I can delete the unused methods in sun/net/util/URLUtil.java and add [JDK-8344180](https://bugs.openjdk.org/browse/JDK-8344180) as additional issue to this PR and handle it all-together. Please let me know if the above works or if you want to handle URLUtil.java changes separately.

Comment (dfuch) [reply src/java.desktop/share/classes/sun/awt/image/URLImageSource.java:48]: Hi @azvegint, if you have no objection to include the deletion of `URLUtil.getConnectPermission()` in this PR, then please do.

**Review by prrace — APPROVED** [review]
If it builds, we should be good.

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**